### ISSUE TITLE
Zero config: Email attribute required after specifying username for sign in

### ISSUE THREAD

**Issue by joebuono** [aws-amplify/amplify-ui#1290]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Create React App

### Please describe your bug.

I went through the [Amplify getting started example](https://docs.amplify.aws/start/getting-started/setup/q/integration/react/), and when I ran `amplify add auth`, I selected:
 `"How do you want users to be able to sign in? Username"`

But when I tried running my app locally and creating a new account, I got the error below. 

I was able to fix it by adding `signUpAttributes={['email']}` to the `<Authenticator />`.

<img width="491" alt="Screen Shot 2022-02-09 at 3 43 03 PM" src="https://user-images.githubusercontent.com/48109584/153295478-f7fa77c9-409d-4e05-835f-cbf6501a448f.png">
 

### What's the expected behaviour?

Zero config should detect that I picked `username` in the set up. 

### Help us reproduce the bug!

1. `create-react-app`
2. Initialize new Amplify project (following [these docs](https://docs.amplify.aws/start/getting-started/setup/q/integration/react/#initialize-a-new-backend))
3. When running `amplify add auth`, select this option: `How do you want users to be able to sign in? Username`
3. In the `App.js` file, copy/paste the code snippet from the section below
4. Attempt to create a new account

### Code Snippet

```typescript
// Put your code below this line.
/* src/App.js */
import { Amplify } from 'aws-amplify';
import { Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css';

import awsExports from './aws-exports';
Amplify.configure(awsExports);

export default function App() {
  return (
    <Authenticator>
      {({ signOut, user }) => (
        <main>
          <h1>Hello {user.username}</h1>
          <button onClick={signOut}>Sign out</button>
        </main>
      )}
    </Authenticator>
  );
}


```

### Additional information and screenshots

`package.json`

```
  "dependencies": {
    "@aws-amplify/ui-react": "^2.3.0",
    "@testing-library/jest-dom": "^5.16.2",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "aws-amplify": "^4.3.14",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "5.0.0",
    "web-vitals": "^2.1.4"
  },
```

**Comment by wlee221** [comment]
Can you provide us `aws-exports.js` with senesitive informations redacted?

**Comment by joebuono** [comment]
Yes, here's `aws-exports.js`. Let me know if I can provide more information. 

```
const awsmobile = {
    "aws_project_region": "us-east-1",
    "aws_appsync_graphqlEndpoint": "[redacted]",
    "aws_appsync_region": "us-east-1",
    "aws_appsync_authenticationType": "API_KEY",
    "aws_appsync_apiKey": "[redacted]",
    "aws_cognito_identity_pool_id": "[redacted]",
    "aws_cognito_region": "us-east-1",
    "aws_user_pools_id": "[redacted]",
    "aws_user_pools_web_client_id": "[redacted]",
    "oauth": {}
};


export default awsmobile;

```

**Comment by wlee221** [comment]
That looks like it doesn't have any zeroconfig information -- it should have additional fields like

```
    "aws_cognito_username_attributes": [],
    "aws_cognito_social_providers": [],
    "aws_cognito_signup_attributes": [
        "PREFERRED_USERNAME"
    ],
    "aws_cognito_mfa_configuration": "OFF",
    "aws_cognito_mfa_types": [
        "SMS"
    ],
    "aws_cognito_password_protection_settings": {
        "passwordPolicyMinLength": 8,
        "passwordPolicyCharacters": [
            "REQUIRES_LOWERCASE",
            "REQUIRES_NUMBERS",
            "REQUIRES_SYMBOLS",
            "REQUIRES_UPPERCASE"
        ]
    },
    "aws_cognito_verification_mechanisms": [
        "EMAIL"
    ]
```

Can you verify you're on latest `@aws-amplify/cli` and try `amplify pull` again?

**Comment by joebuono** [comment]
@wlee221 You're right! I had to update `@aws-amplify/cli` to latest in order for zeroconfig to work properly. I went through the example again and it works as expected. Thanks for your help, closing this issue
